### ISSUE TITLE
Default EarlyStopping strategy

### ISSUE THREAD

**Issue by ml-evs** [ppdebreuck/modnet#45]
As of #44, our strategy is to monitor the loss and stop early by default when performing hyperparameter optimisation. We do not, however, restore the best weights (those with the lowest loss).

Questions:
- should we monitor validation loss by default instead?
- should we restore the best weights for the given preset?
